src/raw/mod.rs

```diff
@@ -46,14 +46,15 @@ unsafe fn offset_from<T>(to: *const T, from: *const T) -> usize {
 // outweighs any gains over the generic implementation.
 #[cfg(all(
     target_feature = "sse2",
-    any(target_arch = "x86", target_arch = "x86_64")
+    any(target_arch = "x86", target_arch = "x86_64"),
+    not(miri)
 ))]
 #[path = "sse2.rs"]
 mod imp;
-#[cfg(not(all(
+#[cfg(any(not(all(
     target_feature = "sse2",
     any(target_arch = "x86", target_arch = "x86_64")
-)))]
+)), miri))]
```

Comment:
I think this would look nicer as:
```rust
#[cfg(not(all(
    target_feature = "sse2",
    any(target_arch = "x86", target_arch = "x86_64"),
    not(miri)
)))]
```
So that it mirrors the condition above.

Reply:
Would it be ok to add cfg_if as a dependency ?

Reply:
No, this crate should not have any dependencies since it is used by libstd (which does funky things with dependency resolution).

Reply:
`cfg-if` is (or could be) `#[no_core]` IIRC, and it is available in libstd:  https://github.com/rust-lang/rust/blob/master/src/libstd/macros.rs#L949 

But I think we pretty much duplicate it on every component we need it (so I can just copy-paste it instead, we do that in libc, stdsimd, and libstd itself).